app/experimenter/visualization/tests/api/test_views.py

```diff
@@ -0,0 +1,69 @@
+import json
+from unittest.mock import patch
+
+from django.test import TestCase
+from django.urls import reverse
+from parameterized import parameterized
+
+from experimenter.experiments.constants import ExperimentConstants
+from experimenter.experiments.tests.factories import ExperimentFactory
+
+
+class TestVisualizationView(TestCase):
+    @parameterized.expand(
+        [
+            ExperimentConstants.STATUS_ACCEPTED,
+            ExperimentConstants.STATUS_COMPLETE,
+        ]
+    )
+    @patch("django.core.files.storage.default_storage.exists")
+    def test_visualization_view_no_data(self, status, mock_exists):
+
+        mock_exists.return_value = False
+        experiment = ExperimentFactory.create_with_status(status)
+
+        response = self.client.get(
+            reverse("visualization", kwargs={"slug": experiment.slug})
+        )
+        self.assertEqual(response.status_code, 200)
+
+        json_data = json.loads(response.content)
+        self.assertEqual({}, json_data)
+
+    @parameterized.expand(
+        [
+            ExperimentConstants.STATUS_ACCEPTED,
+            ExperimentConstants.STATUS_COMPLETE,
+        ]
+    )
+    @patch("django.core.files.storage.default_storage.open")
+    @patch("django.core.files.storage.default_storage.exists")
+    def test_visualization_view_data(self, status, mock_exists, mock_open):
+        DATA_ROW = {"point": 12, "upper": 13, "lower": 10}
+        FULL_DATA = {"daily": DATA_ROW, "weekly": DATA_ROW}
+
+        class file(object):
```

Comment:
[Class names should usually be uppercase](https://www.python.org/dev/peps/pep-0008/#class-names)

Reply:
[Also subclassing `object` isn't necessary anymore, it's implicit in Python 3](https://docs.python.org/3/tutorial/classes.html#multiple-inheritance)